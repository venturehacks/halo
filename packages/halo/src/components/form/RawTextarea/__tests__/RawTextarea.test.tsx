import { render } from 'enzyme';
import React from 'react';

import { RawTextarea } from '../index';

describe('RawTextarea', () => {
  test('smoke', () => {
    const component = render(
      <RawTextarea onChange={() => {}} value="textarea value" />,
    );

    expect(component.text()).toContain('textarea value');
  });

  describe('snapshot', () => {
    describe('size', () => {
      test('sm', () => {
        const component = render(
          <RawTextarea onChange={() => {}} size="sm" value="small" />,
        );
        expect(component).toMatchSnapshot();
      });

      test('md', () => {
        const component = render(
          <RawTextarea onChange={() => {}} value="medium" />,
        );
        expect(component).toMatchSnapshot();
      });

      test('lg', () => {
        const component = render(
          <RawTextarea onChange={() => {}} size="lg" value="large" />,
        );
        expect(component).toMatchSnapshot();
      });
    });
  });
});
