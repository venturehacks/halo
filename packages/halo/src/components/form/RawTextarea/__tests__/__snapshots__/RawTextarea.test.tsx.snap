// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RawTextarea snapshot size lg 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 p-3 h-12 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  large
</textarea>
`;

exports[`RawTextarea snapshot size md 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 px-3 py-2 h-10 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  medium
</textarea>
`;

exports[`RawTextarea snapshot size sm 1`] = `
<textarea
  class="component text-dark-aaaa text-md max-w-full placeholder-dark-a bg-white rounded focus:border-2 px-3 py-1 h-8 border-slate-500 border disabled:text-dark-a disabled:bg-slate-200 disabled:border-light-aa w-auto"
>
  small
</textarea>
`;
